cmd/otel-allocator/main.go

```diff
@@ -81,7 +81,7 @@ func main() {
 	log := ctrl.Log.WithName("allocator")
 
 	allocatorPrehook = prehook.New(cfg.FilterStrategy, log)
-	allocator, err = allocation.New(cfg.AllocationStrategy, log, allocation.WithFilter(allocatorPrehook))
+	allocator, err = allocation.New(cfg.AllocationStrategy, log, allocation.WithFilter(allocatorPrehook), allocation.WithFallbackStrategy(cfg.AllocationFallbackStrategy))
```

Comment:
I'd prefer if we only used allocation.WithFallbackStrategy here if it is actually set. Relying on the function to correctly deal with an empty strategy name feels unnecessarily error-prone.

Reply:
Added conditional here